uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -282,10 +282,17 @@ async def asgi_receive(self):
             return {"type": "websocket.connect"}
 
         await self.handshake_completed_event.wait()
+
+        if self.closed_event.is_set():
+            # If client disconnected, use WebSocketServerProtocol.close_code property.
+            # If the handshake failed or the app closed before handshake completion,
+            # use 1006 Abnormal Closure.
+            return {"type": "websocket.disconnect", "code": self.close_code or 1006}
```

Comment:
looks good to me overall @kylebebak  thanks a lot for that, one small comment.

I noticed that if in `tests.protocols.test_websocket.test_server_reject_connection`, should you add `config = Config(app=app, ws=ws_protocol_cls, http=http_protocol_cls, lifespan="off", log_level="trace")` to get fine-grained output, then the close code is None for the `WSProtocol` and 1006 for the `WebsocketProtocol`, why this is the case, shouldn't both be aligned ?

`WebsocketProtocol`
```
PASSED [100%]INFO:     Started server process [346460]
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
TRACE:    127.0.0.1:49858 - HTTP connection made
TRACE:    127.0.0.1:49858 - Upgrading to WebSocket
DEBUG:    = connection is CONNECTING
TRACE:    127.0.0.1:49858 - WebSocket connection made
DEBUG:    < GET / HTTP/1.1
DEBUG:    < host: 127.0.0.1:8000
DEBUG:    < upgrade: websocket
DEBUG:    < connection: Upgrade
DEBUG:    < sec-websocket-key: DYANNbdkc+2NIoMN3KZIkA==
DEBUG:    < sec-websocket-version: 13
DEBUG:    < sec-websocket-extensions: permessage-deflate; server_max_window_bits=12; client_max_window_bits=12
DEBUG:    < user-agent: Python/3.9 websockets/10.0
TRACE:    127.0.0.1:49858 - ASGI [1] Started scope={'type': 'websocket', 'asgi': {'version': '3.0', 'spec_version': '2.1'}, 'scheme': 'ws', 'server': ('127.0.0.1', 8000), 'client': ('127.0.0.1', 49858), 'root_path': '', 'path': '/', 'raw_path': '/', 'query_string': b'', 'headers': '<...>', 'subprotocols': []}
TRACE:    127.0.0.1:49858 - ASGI [1] Receive {'type': 'websocket.connect'}
TRACE:    127.0.0.1:49858 - ASGI [1] Send {'type': 'websocket.close'}
INFO:     ('127.0.0.1', 49858) - "WebSocket /" 403
DEBUG:    > HTTP/1.1 403 Forbidden
DEBUG:    > Date: Wed, 24 Nov 2021 07:36:20 GMT
DEBUG:    > Server: Python/3.9 websockets/10.0
DEBUG:    > Content-Length: 0
DEBUG:    > Content-Type: text/plain
DEBUG:    > Connection: close
DEBUG:    > [body] (0 bytes)
INFO:     connection failed (403 Forbidden)
DEBUG:    x closing TCP connection
TRACE:    127.0.0.1:49858 - WebSocket connection lost
DEBUG:    = connection is CLOSED
TRACE:    127.0.0.1:49858 - ASGI [1] Receive {'type': 'websocket.disconnect', 'code': 1006}
TRACE:    127.0.0.1:49858 - ASGI [1] Completed
INFO:     connection closed
INFO:     Shutting down
```

 `WSProtocol`
```
PASSED [ 75%]INFO:     Started server process [346460]
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
TRACE:    127.0.0.1:49842 - HTTP connection made
TRACE:    127.0.0.1:49842 - Upgrading to WebSocket
TRACE:    127.0.0.1:49842 - WebSocket connection made
TRACE:    127.0.0.1:49842 - ASGI [1] Started scope={'type': 'websocket', 'asgi': {'version': '3.0', 'spec_version': '2.1'}, 'http_version': '1.1', 'scheme': 'ws', 'server': ('127.0.0.1', 8000), 'client': ('127.0.0.1', 49842), 'root_path': '', 'path': '/', 'raw_path': '/', 'query_string': b'', 'headers': '<...>', 'subprotocols': []}
TRACE:    127.0.0.1:49842 - ASGI [1] Receive {'type': 'websocket.connect'}
TRACE:    127.0.0.1:49842 - ASGI [1] Send {'type': 'websocket.close'}
INFO:     ('127.0.0.1', 49842) - "WebSocket /" 403
TRACE:    127.0.0.1:49842 - ASGI [1] Receive {'type': 'websocket.disconnect', 'code': None}
TRACE:    127.0.0.1:49842 - ASGI [1] Completed
TRACE:    127.0.0.1:49842 - WebSocket connection lost
INFO:     Shutting down
```